index.js

```diff
@@ -1,20 +1,22 @@
 'use strict';
-module.exports = (condition, interval) => new Promise((resolve, reject) => {
-	interval = typeof interval === 'number' ? interval : 20;
+const nonBooleanError = new TypeError('Expected condition to return a boolean');
 
-	const check = () => {
-		Promise.resolve().then(condition).then(val => {
-			if (typeof val !== 'boolean') {
-				throw new TypeError('Expected condition to return a boolean');
-			}
+module.exports = (condition, interval) => {
+	interval = typeof interval === 'number' ? interval : 20;
 
-			if (val === true) {
-				resolve();
-			} else {
-				setTimeout(check, interval);
-			}
-		}).catch(reject);
-	};
+	return new Promise((resolve, reject) => {
+		const check = () => {
+			Promise.resolve(condition()).then(val => {
+				if (val === true) {
+					resolve();
+				} else if (val === false) {
+					setTimeout(check, interval);
+				} else {
+					throw nonBooleanError;
+				}
+			}, reject);
```

Comment:
There's a good reason I used `.catch()` here. If the function in the first argument throws, the second argument will not be called. Which means `nonBooleanError` would just be silently ignored.